backend/ohq/vector_db.py

```diff
@@ -0,0 +1,146 @@
+import openai
+import time
+import tiktoken
+import unidecode
+
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+from ohq.models import Document
+
+openai.api_key = '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'
+encoding = tiktoken.get_encoding("cl100k_base")
+
+pinecone.init(api_key='2a81b230-ba5d-4290-a3ac-b60e6ad17a66', environment='gcp-starter')
+
+index_name = 'ohq-test'
+
+if index_name not in pinecone.list_indexes():
+    # If does not exist, create index
+    pinecone.create_index(
+        index_name,
+        # Dimensionality of text-embedding-ada-002
+        dimension=1536,
+        metric='cosine'
+    )
+    # Wait for index to be initialized
+    time.sleep(1)
+
+index = pinecone.Index(index_name)
+
+"""
+decide which chunking method to use
+"""
+def chunk_text(text):
+    text_splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(
+            chunk_size = 256,
+            chunk_overlap  = 20,
+        )
+    texts = []
+    for document in text_splitter.create_documents([text]):
+        texts.append(document.page_content)
+    return texts
+
+def chunk_text_by_token_count(text, max_tokens=8192):
+    # Initialize the OpenAI GPT tokenizer - this is the one for GPT 3.5
+    tokenizer = tiktoken.get_encoding("cl100k_base")
+
+    current_chunk = ""
+    current_tokens = 0
+
+    for word in text.split(" "):
+        word_tokens = len(list(tokenizer.encode(word)))
+        if current_tokens + word_tokens > max_tokens:
+            yield current_chunk
+            current_chunk = ""
+            current_tokens = 0
+        current_chunk += word + " "
+        current_tokens += word_tokens
+
+    yield current_chunk
+
+def embed_vectors(chunk):
+    response = openai.Embedding.create(model="text-embedding-ada-002", input=chunk)
+    return response["data"][0]["embedding"]
+
+def embed_vector_with_metadata(chunk, metadata):
+    embedding = embed_vectors(chunk)
+    return embedding, metadata
+
+def sanitize_to_ascii(text):
+    ascii_text = unidecode.unidecode(text)
+    # Replace spaces with underscores
+    ascii_text = ascii_text.replace(' ', '_')
+    return ascii_text
+
+def search_index(term, top_k=5):
+    embedded_term = embed_vectors(term)
+
+    pinecone.init(api_key='2a81b230-ba5d-4290-a3ac-b60e6ad17a66', environment='gcp-starter')
+    index = pinecone.Index(index_name)
+
+    query_response = index.query(vector=[embedded_term], top_k=top_k)
+
+    matches = query_response.get('matches', [])
+
+    results = []
+    for match in matches:
+        print(match)
+        print("this is not metadata query")
+        vector_id = match.get('id')
+        score = match.get('score')
+        results.append(f"Vector ID: {vector_id}, Score: {score}")
+
+    return results, len(matches)
+
+def search_index_with_metadata(term, metadata, top_k=5):
+    embedded_term = embed_vectors(term)
+
+    pinecone.init(api_key='2a81b230-ba5d-4290-a3ac-b60e6ad17a66', environment='gcp-starter')
```

Comment:
Same comments as the one about secret keys. Use the global `django.conf` settings object

Reply:
Remove this